src/dataframe/dataframe.jl

```diff
@@ -1812,6 +1812,9 @@ end
 
 # This is not exactly copy! as in general we allow axes to be different
 function _replace_columns!(df::DataFrame, newdf::DataFrame)
+    # here we do not support wastransform argument like for SubDataFrame
+    # because by default in DataFrame case columns are not copied
+    # so we need to pass `:` to select! to handle this case correctly
```

Comment:
I don't get what `select!` call this refers to.

Reply:
`_replace_columns!` is used by `transform!` and `select!`. For `DataFrame` objects `transform!` falls back to `select!` by passing extra `:` (as opposed to what we now do in `GroupedDataFrame` case).

I might drop this comment if you feel it is not useful. This is exactly related to the fact that `select` for `DataFrame` uses a less robust de-aliasing approach (checking input-output mapping) than the approach I proposed for `GroupedDataFrame` with `IdDict`.

Reply:
I have simplified the comment